Perlang.Tests/ConsoleApp/ProgramTest.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections.Generic;
+using Perlang.ConsoleApp;
+using Xunit;
+
+namespace Perlang.Tests.ConsoleApp
+{
+    public class ProgramTest
+    {
+        private readonly Program subject;
+        private readonly List<string> output = new List<string>();
+
+        public ProgramTest()
+        {
+            subject = new Program(standardOutputHandler: s => output.Add(s), runtimeErrorHandler: e => throw e);
+        }
+
+        [Fact]
+        public void Run_can_run_multiple_statements_separated_by_semicolon()
+        {
+            subject.Run("var a = 42; print a;");
+
+            Assert.Equal(new List<string> {"42"}, output);
+        }
+
+        [Fact]
+        public void Run_state_persists_between_invocations()
+        {
+            subject.Run("var a = 42;");
+            subject.Run("print a;");
+
+            Assert.Equal(new List<string> {"42"}, output);
+        }
+
+        [Fact]
+        public void Run_last_variable_defined_with_a_given_name_takes_precedence()
+        {
+            // This is definitely a bug. The second statement should cause an error, since a global variable with the
+            // same name already exists. For now, this test will remember us that these are the currently expected
+            // semantics.
+            subject.Run("var a = 42;");
+            subject.Run("var a = 44;");
+            subject.Run("print a;");
+
+            Assert.Equal(new List<string> {"42"}, output);
+        }
```

Comment:
This is of course not how it _should_ be, but I'm too tired to have the energy to fix it properly right now. We should at least make a proper issue about fixing it sometime.

Reply:
I managed to find and fix this problem, but now I have another one instead... Will write about it separately.